x-pack/plugins/infra/public/pages/logs/log_entry_rate/sections/anomalies/visualisation.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useMemo } from 'react';
+import { AutoRefresh, StringTimeRange } from '../../use_log_entry_rate_results_url_state';
+import { useKibanaContextForPlugin } from '../../../../../hooks/use_kibana';
+import {
+  ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
+  AnomalySwimlaneEmbeddableInput,
+} from '../../../../../../../ml/public';
+import { EmbeddableRenderer } from '../../../../../../../../../src/plugins/embeddable/public';
+import { partitionField } from '../../../../../../common/infra_ml';
+
+interface Props {
+  stringTimeRange: StringTimeRange;
+  jobIds: string[];
+  selectedDatasets: string[];
+  autoRefresh: AutoRefresh;
+}
+
+export const Visualisation: React.FC<Props> = (props) => {
+  const { embeddable: embeddablePlugin } = useKibanaContextForPlugin().services;
+  if (!embeddablePlugin) return null;
+  return <VisualisationContent {...props} />;
+};
+
+export const VisualisationContent: React.FC<Props> = ({
+  stringTimeRange,
+  jobIds,
+  selectedDatasets,
+  autoRefresh,
+}) => {
+  const { embeddable: embeddablePlugin } = useKibanaContextForPlugin().services;
+  const factory: any = embeddablePlugin!.getEmbeddableFactory(ANOMALY_SWIMLANE_EMBEDDABLE_TYPE);
+
+  const refreshConfig = useMemo(() => {
+    const { interval, isPaused } = autoRefresh;
+
+    return {
+      pause: isPaused,
+      value: interval,
+    };
+  }, [autoRefresh]);
+
+  const embeddableInput: AnomalySwimlaneEmbeddableInput = useMemo(() => {
+    return {
+      id: 'LOG_ENTRY_ANOMALIES_EMBEDDABLE_INSTANCE', // NOTE: This is the only embeddable on the anomalies page, a static string will do.
+      jobIds,
```

Comment:
:thinking: Looks like we have to be careful about the job ids we're passing here to avoid this situation when one of the two jobs is missing:

![image](https://user-images.githubusercontent.com/973741/106796703-044b4c00-665c-11eb-9e03-9be9a928e8e4.png)

Reply:
I was also unable to get out if this state by refreshing after the second job has been created. The refresh button of the date picker doesn't seem to have an effect on the embeddable.

Reply:
Ouch. I'll fix this. This is interesting as I'd actually very specifically tested adding a totally fake string in as a Job ID, but alongside the other two real ones. And I remember that didn't show this callout 🤔  Will sort it out though. 

> I was also unable to get out if this state by refreshing after the second job has been created. The refresh button of the date picker doesn't seem to have an effect on the embeddable.

Will also check the state refresh.

Reply:
@Kerry350 the ML Kibana endpoints [check](https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/ml/server/lib/ml_client/ml_client.ts#L54) if provided jobs exist in the current space, so I presume that's why it throws an error.